### PR TITLE
Add support for PF2eMonsterTool import

### PR DESCRIPTION
**Pull request by jayden-hunter**
## Pull Request Description

This Pull Request adds an import option for PF2eMonsterTool, which can be found [here](https://monster.pf2.tools/).

## Changes Proposed

The main changes are as follows:

- [x] Added PF2eMonsterTool Button to Settings Menu
- [x] Added PF2eMonsterTool Parsing
- [x] Sorted Imports in `src/importers/index.ts` and `src/importers/importer.worker.ts` into alphabetical order. (my linter did that)

## Related Issues

No related issues. Can create one if necessary.

## Checklist

- [x] I have read the contribution guidelines and code of conduct.
- [x] I have tested the changes locally and they are working as expected.
- [x] I have added appropriate comments and documentation for the code changes.
- [x] My code follows the coding style and standards of this project.
- [x] I have rebased my branch on the latest main (or master) branch.
- [x] All tests (if applicable) have passed successfully.
- [x] I have run linters and fixed any issues.
- [x] I have checked for any potential security issues or vulnerabilities.

## Screenshots (if applicable)

image

## Additional Notes

This Plugin has been so useful to me, thank you very kindly for all of the work you've done on it!

### CODE REVIEW

File: src/importers/pf2eMonsterToolImport.ts [hunk 1]
```diff
@@ -0,0 +1,267 @@
...
+                    ac: ac,
+                    armorclass: getACStats(ac, getValue(json.fortitude), getValue(json.reflex), getValue(json.will)),
+                    hp: hp,
+                    health: getHealthStats(hp, json.immunity, json.resistance, json.weakness),
+                    attacks: getAttacks(json.strikes),
+                    spellcasting: spellcasting,
+                    speed: json.speed || "5 feet",
+                    stats: stats,
+                    abilityMods: stats,
+                    damage_vulnerabilities: json.weakness,
```
Comment (miscoined) [inline src/importers/pf2eMonsterToolImport.ts:48]: I don't think these are used outside of the 5e layout.

**Comment by miscoined** [comment]
Just in case it's worth mentioning - if #438 gets merged in, that layout would probably be easier to import into without as much data wrangling.

This is great by the way, nice work :)

**Comment by valentine195** [comment]
I will try to review this within the next week. 

As an aside, this PR made me start thinking about creating a base importer class, to make it easier to PR new ones onto. I also want to rework the settings UI, it is getting quite cluttered.

File: src/importers/pf2eMonsterToolImport.ts (hunk 1 above)
Comment (jayden-hunter) [reply src/importers/pf2eMonsterToolImport.ts:48]: Yeah, I mostly just added those in to conform to the `Monster` type, which requires those parameters. It didn't bother me much that they were required/vestigial, I was assuming they were mostly for searching/filtering, so filled them in just to be safe. It'd be very easy just to leave them as empty strings as well, which wouldn't break anything from what I understand.

**Comment by jayden-hunter** [comment]
> Just in case it's worth mentioning - if #438 gets merged in, that layout would probably be easier to import into without as much data wrangling.

If it does, I'll be happy to update the importer :)

**Comment by jayden-hunter** [comment]
Just made a couple changes to the logic, clearly I hadn't tested extensively enough. Should be working now, sorry.
Presuming there'll be some other things I'll have missed too, but happy to fix anything if I've screwed it up 👍

File: src/importers/pf2eMonsterToolImport.ts (hunk 1 above)
Comment (valentine195) [reply src/importers/pf2eMonsterToolImport.ts:47]: > which requires those parameters. It didn't bother me much that they were required/vestigial

yeah... these are vestigial. the whole `Monster` type became almost functionally useless after extending beyond 5e, I just never got around to refactoring it out.

**Pull request merged**